knowledge-base/implementation-documentation/0005-api-improvements.md

```diff
@@ -214,6 +214,54 @@ Cons:
 - Difficult to implement.
 - Not realistic for 2.0, but maybe later.
 
+## Function style filtering [Updated]
+
+The new filter api is implemented as non-breaking change and the current methods are deprecated. Lambda filters on collection properties will be implemented as a new feature in the major version.
+
+### Current API
+
+```ts
+peopleApi.requestBuilder()
+  .getAll()
+  .filter(
+    peopleApi.schema.USER_NAME.equals('Peter'),
+    peopleApi.schema.PHOTOS.filter(photoApi.schema.NAME.equals('Photo1'))
+    peopleApi.schema.TO_FRIENDS.filter(
+      any(
+        peopleApi.schema.FIRST_NAME.equals('John'),
+        peopleApi.schema.LAST_NAME.equals('Miller')
+      )
+    )
+  );
+```
+
+### Examples
+
+**Option 1: Similar to the current api**
+
+```ts
+.filter(({ USER_NAME, PHOTOS, TO_FRIENDS, EMAILS, ADDRESSES }) => and(
```

Comment:
[req] In general, the .md files are not scanned by lint, so maybe you have to do it manually;)